composer/core/state.py

```diff
@@ -944,6 +944,12 @@ def load_state_dict(
                 self._load_dataset_state(serialized_value)
             elif attribute_name == 'optimizers':
                 self.load_optim_state(state)
+            elif attribute_name in ['train_metrics', 'eval_metrics']:
+                state_field_value = getattr(self, attribute_name)
+                # Update the metrics with the serialized values
+                for metric_name, metric in serialized_value.items():
```

Comment:
Eval metrics have are actually a dict of dicts, so another layer of unwrapping is needed (that is what is causing the test failure)